### ISSUE TITLE
[Question] How to discard a group after use?

### ISSUE THREAD

**Issue by neo-mashiro** [skypjack/entt#770]
Hi, another beginner's question here~ I'm trying to use groups inside a function, where I need to process some combinations of components using multiple groups, and I got a runtime assertion error.
```c++
void Foo() {
    ....
    
    if (auto group = registry.group<A, B>(); !group.empty()) {
        auto& b = group.get<B>(group.front());
        ....
    }

    if (ctx.lightmask.test(0)) {
        auto group = registry.group<C, A>();
        group.sort([](const entt::entity a, const entt::entity b) { return a < b; });
        ....
    }
}
```
As per the documentation, these are conflicting groups and the second one is going to fail, so before creating the second group, I need to somehow discard the first group, but I couldn't find a function for that. I tried to wrap them inside an if-else block so that each group has a local scope, but it seems that the first group is still alive in the registry after it goes out of scope, I wonder if there's a way to manually clean it up? or perhaps the way I'm using groups is not correct?

**Comment by skypjack** [comment]
Currently, you cannot and it's mostly on purpose. Groups aren't meant to be built and destroyed continuously at runtime. Quite the opposite actually, they are meant to optimize a specific access pattern in the long run.

**Comment by neo-mashiro** [comment]
> 
> 
> they are meant to optimize a specific access pattern in the long run.

OK I got your point, so a group is like a rearranged owning subset of the registry, which is somewhat analogous to the RAM of CPU. Groups are intended to be setup once and used repeatedly, where later access can be made blazingly fast, but views do not take ownership so can be considered more "temporary" than groups. Is my understanding correct?

I've re-read the groups part in the documentation and it's now making more sense to me. I removed the ownership of A by using `registry.group<B>(A)` and `registry.group<C>(A)` and the assertion passed, thanks for your clarification!
